backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -312,7 +313,6 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
             LOGGER);
         repository.update(request);
       }
-      repository.setDeleted(true);
```

Comment:
On update case, it will mark repository as deleted which is wrong so move it to up in `if (create) {` section.